### PR TITLE
📚Switch to using DocumenterInterLinks.jl

### PR DESCRIPTION
**Pull request by kellertuer**
Switches ManifoldsBase links to use DocumenterInterLinks

### CODE REVIEW

File: .github/workflows/documenter.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ jobs:
           version: 1.3.361
       - uses: julia-actions/setup-julia@latest
         with:
-          version: 1.9
+          version: 1.10
```
Comment (mateuszbaran) [inline .github/workflows/documenter.yml:20]:
```suggestion
          version: "1.10"
```
Without `""` YAML considers it to be a floating point number equal to 1.1.

Comment (kellertuer) [reply .github/workflows/documenter.yml]: Ah yes debugged for quite a while the same issue on Manopt and was not careful enough yesterday.

**Comment by kellertuer** [comment]
Ah, we seem to have the same problem with BVPs here as well.

**Comment by mateuszbaran** [comment]
I've noticed that it was broken by NonlinearSolve v 3.8 so I've upper-bounded it in my PR. Hopefully CI picks the right version.

**Comment by kellertuer** [comment]
Sue, still a bit annoying that they again broke their package

**Comment by mateuszbaran** [comment]
And CI doesn't respect that upper bound... I've asked on Slack how to upper-bound it.

File: src/manifolds/VectorBundle.jl [hunk 2]
```diff
@@ -20,7 +20,7 @@ fiber_bundle_transport(M::AbstractManifold, ::FiberType) =
     VectorBundle{𝔽,TVS,TM,VTV} = FiberBundle{𝔽,TVS,TM,TVT} where {TVS<:VectorSpaceType}
 
 Alias for [`FiberBundle`](@ref) when fiber type is a `TVS` of type
-[`VectorSpaceType`]https://juliamanifolds.github.io/ManifoldsBase.jl/stable/bases/#ManifoldsBase.VectorSpaceType).
+[`VectorSpaceType`]@extref `ManifoldsBase.VectorSpaceType`).
```
Comment (mateuszbaran) [inline src/manifolds/VectorBundle.jl:23]:
```suggestion
[`VectorSpaceType`](@extref `ManifoldsBase.VectorSpaceType`).
```

**Review by mateuszbaran — APPROVED** [review]

**Pull request merged**